props/saber_sa22c_buttons.h

```diff
@@ -0,0 +1,334 @@
+#ifndef DISABLE_COLOR_CHANGE
+        if (accel_.x < -0.15) {
+          ToggleColorChangeMode();
+        } else {
+          SaberBase::DoForce();
+        }
+#else
+        SaberBase::DoForce();
+#endif
+        return true;
+      case EVENTID(BUTTON_POWER, EVENT_CLICK_LONG, MODE_ON):
+        StartOrStopTrack();
+        return true;
+      case EVENTID(BUTTON_POWER, EVENT_CLICK_SHORT, MODE_ON):
+        SaberBase::DoBlast();
+        return true;
+#endif
+      case EVENTID(BUTTON_POWER, EVENT_DOUBLE_CLICK, MODE_ON):
+        if (millis() - activated_ < 500) {
+          if (SetMute(true)) {
+            unmute_on_deactivation_ = true;
+          }
+        }
+#if NUM_BUTTONS > 1
+        else {
+          SaberBase::DoForce();
+        }
+#endif
+        return true;
+
+#if NUM_BUTTONS == 1
+        case EVENTID(BUTTON_POWER, EVENT_HELD_LONG, MODE_ON):
+        if (!SaberBase::Lockup()) {
+#ifndef DISABLE_COLOR_CHANGE
+          if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) {
+            // Just exit color change mode.
+            // Don't turn saber off.
+            ToggleColorChangeMode();
+            return true;
+          }
+#endif
+          Off();
+        }
+        return true;
+#endif
+
+#ifndef DISABLE_COLOR_CHANGE
+#ifdef NO_LOCKUP_HOLD
+    case EVENTID(BUTTON_POWER, EVENT_CLICK_SHORT, MODE_ON | BUTTON_AUX):
+      ToggleColorChangeMode();
+      break;
+#else //NO_LOCKUP_HOLD
+    case EVENTID(BUTTON_AUX, EVENT_CLICK_SHORT, MODE_ON | BUTTON_POWER):
+      ToggleColorChangeMode();
+      break;
+#endif //NO_LOCKUP_HOLD
+#endif //DISABLE_COLOR_CHANGE
+#if NUM_BUTTONS > 1
+      case EVENTID(BUTTON_POWER, EVENT_HELD_MEDIUM, MODE_ON):
+      case EVENTID(BUTTON_POWER, EVENT_LATCH_OFF, MODE_ON):
+      case EVENTID(BUTTON_AUX, EVENT_LATCH_OFF, MODE_ON):
+      case EVENTID(BUTTON_AUX2, EVENT_LATCH_OFF, MODE_ON):
+#endif
+#if NUM_BUTTONS == 0
+      case EVENTID(BUTTON_NONE, EVENT_TWIST, MODE_ON):
+#endif
+#if NUM_BUTTONS != 1
+        if (!SaberBase::Lockup()) {
+#ifndef DISABLE_COLOR_CHANGE
+          if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) {
+            // Just exit color change mode.
+            // Don't turn saber off.
+            ToggleColorChangeMode();
+            return true;
+          }
+#endif
+          Off();
+        }
+#endif
+        return true;
+
+    case EVENTID(BUTTON_NONE, EVENT_CLASH, MODE_ON | BUTTON_AUX):
+    case EVENTID(BUTTON_NONE, EVENT_CLASH, MODE_ON | BUTTON_POWER):
+#ifndef NO_LOCKUP_HOLD
+    case EVENTID(BUTTON_AUX, EVENT_HELD, MODE_ON):
+#endif
+      if (!SaberBase::Lockup()) {
+        if (pointing_down_) {
+          SaberBase::SetLockup(SaberBase::LOCKUP_DRAG);
+        } else {
+          SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL);
+        }
+        SaberBase::DoBeginLockup();
+        return true;
+      }
+
+        // Off functions
+      case EVENTID(BUTTON_POWER, EVENT_CLICK_LONG, MODE_OFF):
```

Comment:
this case (and most of the ones below) are not indented the same as the cases above.